tests/commands/create_project.py

```diff
@@ -0,0 +1,269 @@
+        "--preset-mapping=gnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.report_email == "foo@bar.com"
+
+    with pytest.raises(CommandError):
+        # invalid email address
+        call_command(
+            "create_project",
+            "foo0",
+            "--report-email=foo@bar",
+            "--preset-mapping=gnu")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_source_language(language0):
+    call_command(
+        "create_project",
+        "foo0",
+        "--preset-mapping=gnu",
+        "--source-language=%s" % language0.code)
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.source_language == language0
+
+    with pytest.raises(CommandError):
+        call_command(
+            "create_project",
+            "foo0",
+            "--preset-mapping=gnu",
+            "--source-language=DOESNOTEXIST")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_disabled():
+    call_command(
+        "create_project",
+        "foo0",
+        "--preset-mapping=gnu",
+        "--disabled")
+    foo0 = Project.objects.get(code="foo0")
+    assert foo0.disabled is True
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_preset_mapping():
+    call_command(
+        "create_project",
+        "foo0",
+        "--preset-mapping=nongnu")
+    foo0 = Project.objects.get(code="foo0")
+    assert (
+        foo0.config["pootle_fs.translation_mapping"]
+        == dict(default=u'/<language_code>/<dir_path>/<filename>.<ext>'))
+    with pytest.raises(CommandError):
+        call_command(
+            "create_project",
+            "foo1",
+            "--preset-mapping=gnu",
+            "--mapping=u'/<language_code>/<dir_path>/<filename>.<ext>'")
+        call_command(
+            "create_project",
+            "foo1")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_mapping():
+    call_command(
+        "create_project",
+        "foo0",
+        "--mapping=/<language_code>/<dir_path>/<filename>.<ext>")
+    foo0 = Project.objects.get(code="foo0")
+    assert (
+        foo0.config["pootle_fs.translation_mapping"]
+        == dict(default=u'/<language_code>/<dir_path>/<filename>.<ext>'))
+
+    # This should fail...
+    # with pytest.raises(CommandError):
+    #     call_command(
+    #         "create_project",
+    #         "foo1",
+    #         "--mapping=NOTAMAPPING")
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_cmd_create_project_fs_type():
+
+    class DummyPlugin(Plugin):
+        pass
+
+    @provider(fs_plugins)
+    def dummy_fs_plugin(**kwargs):
+        return dict(dummyfs=DummyPlugin)
+
+    call_command(
+        "create_project",
+        "foo0",
+        "--preset-mapping=gnu",
+        "--fs-type=dummyfs")
+    foo0 = Project.objects.get(code="foo0")
+    assert (
+        foo0.config["pootle_fs.fs_type"]
+        == "dummyfs")
+
+    with pytest.raises(CommandError):
```

Comment:
As with the other `pytest.raises(CommandError)` I would check for the exception message to ensure it fails for the reason we are testing.